### ISSUE TITLE
NullPointerException in class ImplementationValidator

### ISSUE THREAD

**Issue by tbl0605** [pdt-eg/Core-Plugin#138]
Hi,
sometimes I get following error :

java.lang.NullPointerException
	at org.pdtextensions.semanticanalysis.validation.validator.ImplementationValidator.visit(ImplementationValidator.java:110)
	at org.eclipse.php.internal.core.compiler.ast.visitor.PHPASTVisitor.visit(PHPASTVisitor.java:1353)
	at org.eclipse.dltk.ast.declarations.TypeDeclaration.traverse(TypeDeclaration.java:151)
	at org.eclipse.dltk.ast.statements.Block.traverse(Block.java:43)
	at org.eclipse.dltk.ast.declarations.ModuleDeclaration.traverse(ModuleDeclaration.java:69)
	at org.pdtextensions.semanticanalysis.validation.AbstractValidator.validate(AbstractValidator.java:31)
	at org.pdtextensions.semanticanalysis.validation.validator.ImplementationValidator.validate(ImplementationValidator.java:221)
	at org.pdtextensions.internal.semanticanalysis.integration.BuildParticipant.validate(BuildParticipant.java:102)
	at org.pdtextensions.internal.semanticanalysis.integration.BuildParticipant.build(BuildParticipant.java:72)

It's mainly due to the fact that PDT allows reserved keywords as interface names ( see https://bugs.eclipse.org/bugs/show_bug.cgi?id=443037 ), so following example will throw a NPE:
<?php
class self implements parent
{
    function a()
    {
        self::t;
    }
}

So basically it's a PDT bug but we can still correct the NPE thrown by Core-Plugin ;)

You'll find the patch as attachment.

Thierry.

[Core-Plugin.patch.txt](https://github.com/pdt-eg/Core-Plugin/files/247133/Core-Plugin.patch.txt)

**Comment by zulus** [comment]
Can you prepare pulk request?

**Comment by tbl0605** [comment]
Done ;)

**Comment by zulus** [comment]
Merged thanks!
